Z/Data/Parser/Base.hs

```diff
@@ -666,12 +666,30 @@ skipWhile p =
                     in if V.null rest then Partial (go s) else k s () rest
         in go s0
 
+-- | Similar to 'skipWhile', but requires the predicate to succeed on at least one byte
+-- of input: it will fail if the predicate never returns 'True' or reach the end of input
+--
+skipWhile1 :: (Word8 -> Bool) -> Parser ()
+skipWhile1 p = do
+    bs <- takeWhile p
+    if V.null bs
+    then Parser $ \ kf _ _ inp ->
+            kf ["Z.Data.Parser.Base.skipWhile1: no satisfied byte at " <> T.toText (V.take 10 inp) ]
+               inp
+    else pure ()
+
 -- | Skip over white space using 'isSpace'.
 --
 skipSpaces :: Parser ()
 {-# INLINE skipSpaces #-}
 skipSpaces = skipWhile isSpace
 
+-- | Skip over white space and @\'\\n\'@ using 'isSpace'.
+--
+skipSpaces' :: Parser ()
+{-# INLINE skipSpaces' #-}
+skipSpaces' = skipWhile (\ w -> isSpace w || w == NEWLINE)
```

Comment:
Not sure if it worth adding, but only NEWLINE not CARRIAGE_RETURN?

Reply:
i will mv it to benchmark repo